src/Service/CallService.php

```diff
@@ -244,9 +244,16 @@ public function call(
             $config['headers'] = [];
         }
 
+        $url = $source->getLocation().$endpoint;
+
         // Set authentication if needed.
         $createCertificates && $this->getCertificate($config);
-        $config = array_merge_recursive($this->getAuthentication($source, $config), $config);
+        $requestInfo = [
+            'url'    => $url,
+            'method' => $method,
+            'body'   => ($config['body'] ?? ""),
```

Comment:
We already send $config array to the getAuthentication function, it feels weird to also do this^
I understand why you do this, but it might look better if you use the $config array instead in the getHmacToken function 🤔

Reply:
Changed a bit but cant change it totally becuase $config does not get passed to getHmacToken